styx-scheduler-service/src/main/java/com/spotify/styx/StyxScheduler.java

```diff
@@ -451,7 +451,8 @@ private TriggerListener trigger(
 
     return (workflow, triggerId, instant) -> {
       try {
-        if (!storage.enabled(workflow.id()) || !storage.globalEnabled()) {
+        final Set<WorkflowId> enabledWorkflows = storage.enabledWorkflows();
```

Comment:
Unrelated change? Why get all the enabled workflows instead of just checking for a specific one?